phoenix-core/src/main/java/org/apache/phoenix/execute/LiteralResultIterationPlan.java

```diff
@@ -85,7 +85,8 @@ public void close() throws SQLException {
 
             @Override
             public Tuple next() throws SQLException {
-                while (!this.closed && (offset != null && count++ < offset) && tupleIterator.hasNext()) {
+                while (!this.closed && (offset != null && count < offset) && tupleIterator.hasNext()) {
+                    count++;
```

Comment:
Ah, I was in a notion that I updated limit to limit+offset already.
Thanks for catching it, I have corrected it now.